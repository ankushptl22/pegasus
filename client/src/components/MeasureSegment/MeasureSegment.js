import React, { Component } from 'react';
import { Segment, Statistic, Icon, Grid } from 'semantic-ui-react';
import './MeasureSegment.less';

class MeasureSegment extends Component {
    render() {
        const { data } = this.props.measureProp;
        const measures = data.map((measuresData, index) => {
            return (
                <Grid key={index} className="singleGrid">
                    <Statistic>
                        <p>{measuresData.label}</p>
                        <Statistic.Value text >{measuresData.percentage}%</Statistic.Value>
                    </Statistic>
                </Grid>
            );
        });

        return <Grid className="gridCollection">
            <Grid className="singleGrid">
                <Icon size='huge' name={this.props.measureProp.arrow} color='green' />
                <Icon size='huge' name={this.props.measureProp.icon} color='green' />
            </Grid>
            {measures}
        </Grid>;
    }
}

MeasureSegment.defaultProps = {
    measureProp: {
        data: [
            {
                label: "Achieved Performance",
                percentage: "78.08"
            },
            {
                label: "CMS Benchmark",
                percentage: "45"
            },
            {
                label: "Registry Average",
                percentage: "85"
            }
        ],
        icon: "thumbs up outline",
        arrow: "arrow up"
    }
}
export default MeasureSegment;