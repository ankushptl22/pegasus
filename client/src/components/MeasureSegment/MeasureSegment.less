p {
  color: #a2a2a2;
}
.ui.grid.gridCollection {
  box-shadow: none !important;
  margin-bottom: 20px !important;
}
.ui.grid.gridCollection >.ui.grid.singleGrid {
  border-right: 1px solid #a2a2a2;
  margin-left: 3px;
  height: 74px;
}
.ui.grid.gridCollection >.ui.grid.singleGrid:last-child {
  border-right: 0px !important;
}

.ui.grid.gridCollection >.ui.grid.singleGrid:first-child {
  margin-top: 13px;
}

.ui.statistic>.text.value, .ui.statistics .statistic>.text.value {
  line-height: 1em;
  min-height: 2em;
  font-weight: 700;
  text-align: left !important;
  color: #a2a2a2 !important;
}