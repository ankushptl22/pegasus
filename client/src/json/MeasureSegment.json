{
    "measureProp": {
        "data": [
            {
                "label": "Achieved Performance",
                "percentage": "78.08"
            },
            { 
                "label": "CMS Benchmark",
                "percentage": "45"
            },
            {
                "label": "Registry Average",
                "percentage": "85"
            }
        ],
        "icon": "thumbs up outline",
        "arrow": "arrow up"
    }
}