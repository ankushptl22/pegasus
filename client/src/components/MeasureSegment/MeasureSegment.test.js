import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MeasureSegment from '../../components/MeasureSegment/MeasureSegment';
import MeasurePropsData from '../../json/MeasureSegment.json';


Enzyme.configure({ adapter: new Adapter() });
console.log(MeasurePropsData); 
describe('<MeasureSegment />', () => {
    test('renders', () => {
        const wrapper = shallow(<MeasureSegment measureProp={MeasurePropsData.measureProp} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('will check props', () => {
        const component = shallow(<MeasureSegment measureProp={MeasurePropsData.measureProp} />).props();
        expect(component).toBeDefined();
    });
});
