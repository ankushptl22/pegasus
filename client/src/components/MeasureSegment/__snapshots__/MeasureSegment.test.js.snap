// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeasureSegment /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MeasureSegment
    measureProp={
      Object {
        "arrow": "arrow up",
        "data": Array [
          Object {
            "label": "Achieved Performance",
            "percentage": "78.08",
          },
          Object {
            "label": "CMS Benchmark",
            "percentage": "45",
          },
          Object {
            "label": "Registry Average",
            "percentage": "85",
          },
        ],
        "icon": "thumbs up outline",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Segment>
          <Icon
            as="i"
            color="green"
            name="arrow up"
            size="huge"
          />
          <Icon
            as="i"
            color="green"
            name="thumbs up outline"
            size="huge"
          />
        </Segment>,
        Array [
          <Segment>
            <Statistic>
              <p>
                Achieved Performance
              </p>
              <StatisticValue
                text={true}
              >
                78.08
                %
              </StatisticValue>
            </Statistic>
          </Segment>,
          <Segment>
            <Statistic>
              <p>
                CMS Benchmark
              </p>
              <StatisticValue
                text={true}
              >
                45
                %
              </StatisticValue>
            </Statistic>
          </Segment>,
          <Segment>
            <Statistic>
              <p>
                Registry Average
              </p>
              <StatisticValue
                text={true}
              >
                85
                %
              </StatisticValue>
            </Statistic>
          </Segment>,
        ],
      ],
      "className": "seg",
      "horizontal": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Icon
              as="i"
              color="green"
              name="arrow up"
              size="huge"
            />,
            <Icon
              as="i"
              color="green"
              name="thumbs up outline"
              size="huge"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "color": "green",
              "name": "arrow up",
              "size": "huge",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "color": "green",
              "name": "thumbs up outline",
              "size": "huge",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": <Statistic>
            <p>
              Achieved Performance
            </p>
            <StatisticValue
              text={true}
            >
              78.08
              %
            </StatisticValue>
          </Statistic>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Achieved Performance
              </p>,
              <StatisticValue
                text={true}
              >
                78.08
                %
              </StatisticValue>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Achieved Performance",
              },
              "ref": null,
              "rendered": "Achieved Performance",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "78.08",
                  "%",
                ],
                "text": true,
              },
              "ref": null,
              "rendered": Array [
                "78.08",
                "%",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": <Statistic>
            <p>
              CMS Benchmark
            </p>
            <StatisticValue
              text={true}
            >
              45
              %
            </StatisticValue>
          </Statistic>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                CMS Benchmark
              </p>,
              <StatisticValue
                text={true}
              >
                45
                %
              </StatisticValue>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CMS Benchmark",
              },
              "ref": null,
              "rendered": "CMS Benchmark",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "45",
                  "%",
                ],
                "text": true,
              },
              "ref": null,
              "rendered": Array [
                "45",
                "%",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "children": <Statistic>
            <p>
              Registry Average
            </p>
            <StatisticValue
              text={true}
            >
              85
              %
            </StatisticValue>
          </Statistic>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Registry Average
              </p>,
              <StatisticValue
                text={true}
              >
                85
                %
              </StatisticValue>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Registry Average",
              },
              "ref": null,
              "rendered": "Registry Average",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "85",
                  "%",
                ],
                "text": true,
              },
              "ref": null,
              "rendered": Array [
                "85",
                "%",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Segment>
            <Icon
              as="i"
              color="green"
              name="arrow up"
              size="huge"
            />
            <Icon
              as="i"
              color="green"
              name="thumbs up outline"
              size="huge"
            />
          </Segment>,
          Array [
            <Segment>
              <Statistic>
                <p>
                  Achieved Performance
                </p>
                <StatisticValue
                  text={true}
                >
                  78.08
                  %
                </StatisticValue>
              </Statistic>
            </Segment>,
            <Segment>
              <Statistic>
                <p>
                  CMS Benchmark
                </p>
                <StatisticValue
                  text={true}
                >
                  45
                  %
                </StatisticValue>
              </Statistic>
            </Segment>,
            <Segment>
              <Statistic>
                <p>
                  Registry Average
                </p>
                <StatisticValue
                  text={true}
                >
                  85
                  %
                </StatisticValue>
              </Statistic>
            </Segment>,
          ],
        ],
        "className": "seg",
        "horizontal": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Icon
                as="i"
                color="green"
                name="arrow up"
                size="huge"
              />,
              <Icon
                as="i"
                color="green"
                name="thumbs up outline"
                size="huge"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "color": "green",
                "name": "arrow up",
                "size": "huge",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "color": "green",
                "name": "thumbs up outline",
                "size": "huge",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": <Statistic>
              <p>
                Achieved Performance
              </p>
              <StatisticValue
                text={true}
              >
                78.08
                %
              </StatisticValue>
            </Statistic>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  Achieved Performance
                </p>,
                <StatisticValue
                  text={true}
                >
                  78.08
                  %
                </StatisticValue>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Achieved Performance",
                },
                "ref": null,
                "rendered": "Achieved Performance",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "78.08",
                    "%",
                  ],
                  "text": true,
                },
                "ref": null,
                "rendered": Array [
                  "78.08",
                  "%",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": <Statistic>
              <p>
                CMS Benchmark
              </p>
              <StatisticValue
                text={true}
              >
                45
                %
              </StatisticValue>
            </Statistic>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  CMS Benchmark
                </p>,
                <StatisticValue
                  text={true}
                >
                  45
                  %
                </StatisticValue>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CMS Benchmark",
                },
                "ref": null,
                "rendered": "CMS Benchmark",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "45",
                    "%",
                  ],
                  "text": true,
                },
                "ref": null,
                "rendered": Array [
                  "45",
                  "%",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "children": <Statistic>
              <p>
                Registry Average
              </p>
              <StatisticValue
                text={true}
              >
                85
                %
              </StatisticValue>
            </Statistic>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  Registry Average
                </p>,
                <StatisticValue
                  text={true}
                >
                  85
                  %
                </StatisticValue>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Registry Average",
                },
                "ref": null,
                "rendered": "Registry Average",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "85",
                    "%",
                  ],
                  "text": true,
                },
                "ref": null,
                "rendered": Array [
                  "85",
                  "%",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
